### PR TITLE
Update gem dependencies and make it work with spree 3.2.0

### PR DESCRIPTION
**Pull request by k3rn31**
I've set up Apprisals, and made minor fixes to make this gem work with spree 3.2.0

### CODE REVIEW

File: app/views/spree/admin/translations/_form_fields.html.erb [hunk 1]
```diff
@@ -10,21 +10,29 @@
               <div class="pull-right text-muted">
                 <small><i><%= Spree.t(:'i18n.this_file_language') %></i></small>
 
-                <% if locale.to_s.include?('-') %>
-                  <img src="https://www.localeapp.com/assets/flags/regions/<%= locale.to_s.split('-').last %>.png">
-                <% else %>
-                  <img src="https://www.localeapp.com/assets/flags/languages/<%= locale %>.png">
-                <% end %>
+                <% if false %> <%# Start comment out %>
+                  <%# localeapp.com icons don't work anymore %>
```
Comment (mvz) [inline app/views/spree/admin/translations/_form_fields.html.erb]: If this should broken i think it should be removed entirely.

**Comment by mvz** [comment]
Hi @k3rn31 thanks for working on spree_globalize. I would love to see it maintained again. Can you take a look at the Travis failures so this can be merged?

**Comment by k3rn31** [comment]
Hi @mvz, I have all the tests pass on my local machine. It looks like all the failing tests are on PostgreSQL (tests on mySQL are passing). I will take a look into it.

**Comment by k3rn31** [comment]
@mvz, the build fails because of timeouts in fetching gems, and random fails. I don't know why it happens, it looks like a problem with travis. 
I have successful tests on postgres on my local machine.

**Comment by k3rn31** [comment]
@mvz can you try to re-run the build on travis? It fails due to timeouts in fetching gems. I have all test passing on postgres on my local machine.

**Comment by mvz** [comment]
@k3rn31 I'll take a look.

**Comment by mvz** [comment]
It looks like the green specs with mysql are not actually working: When you look at the logs, you can see that it complains of 'Gem::LoadError: Specified 'mysql2' for database adapter'.

I also have no Travis powers (I'm just a visitor), maybe @mauazua can tell Travis to re-run the specs?

**Comment by mvz** [comment]
Maybe you're doing too much in this pull request, since you're also changing the version of globalize used. This makes it harder to get everything working, especially since the build on current master is already failing. I would recommend first making that build succeed and then go from there.

**Comment by k3rn31** [comment]
What's strange is that tests ar failing randomly on travis, and on my local
machine they pass flawlessly.

-- 
Davide Petilli

Il giorno 19 marzo 2017 @ 19:34:30, Matijs van Zuijlen (
notifications@github.com) ha scritto:

**Comment by dsawa** [comment]
@k3rn31 even with your branch, it seems that I still can't use spree_globalize with spree 3.2 and rails 5.

```
    spree_globalize was resolved to 3.2.0.alpha, which depends on
      globalize was resolved to 3.0.0, which depends on
        paper_trail (~> 2) was resolved to 2.0.0, which depends on
          rails (~> 3)
```

paper_trail dependency breaks `bundle_update`

**Comment by k3rn31** [comment]
There's something wrong with the globalize dependency. It should be globalize 5.0
I'll take a look into it later.  

--  
K3rn31  

Il giorno 31 marzo 2017 @ 11:09:56, Dorian Sawa (notifications@github.com(mailto:notifications@github.com)) ha scritto:

**Comment by dsawa** [comment]
Your branch is not working for me with spree 3.2 and rails 5. (globalize dependency conflict).

However if I add to Gemfile something like this:
```
gem 'globalize', '~> 5.1.0.beta1'
gem 'activemodel-serializers-xml' # New globalize dependency
gem 'spree_globalize', github: 'k3rn31/spree_globalize', branch: '
```
it works. 

The disadvantage is however a lot of deprecation warnings coming out from spree tables.
```
DEPRECATION WARNING: You have defined one or more translated attributes with names that conflict with column(s) on the model table. Globalize does not support this configuration anymore, remove or rename column(s) on the model table.
Model name (table name): Spree::Product (spree_products)
Attribute name(s): name, description, meta_description, meta_keywords, slug
 (called from block in <module:Spree> at /home/dsawa/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/bundler/gems/spree_globalize-5d0df498c207/app/models/spree/product_decorator.rb:3)
```

I found out comments about this here: https://github.com/globalize/globalize/issues/562#issuecomment-251920587

However there is still nothing about that in globalize readme. 

Do you have same deprecation warnings on your machine?

**Comment by k3rn31** [comment]
Yes, I have the same deprecation warnings. I tried to fix them, but it's not trivial.  
Besides the warnings, it should work for now.

--  
K3rn31  

Il giorno 5 aprile 2017 @ 13:08:25, Dorian Sawa (notifications@github.com(mailto:notifications@github.com)) ha scritto:

**Comment by nicolasblanco** [comment]
Thanks @k3rn31 for your work as I was able to use your fork with @dsawa Gemfile modifications.

Very strange though, it was failing on my production environment only on this line https://github.com/k3rn31/spree_globalize/blob/master/app/controllers/concerns/spree_globalize/controller_globalize_helper.rb#L14 because of the Fallbacks constant not being available.

I had to manually `require "spree_globalize/fallbacks"` in my initializer to make it work.

**Comment by damianlegawiec** [comment]
Done via https://github.com/spree-contrib/spree_globalize/pull/33

**Pull request closed without merging**